Description: Warn and return without error if /boot is a dm-crypt device
 With any luck the administrator knows what they're doing; in any event, we
 probably can't improve matters by having update-grub exit non-zero.
Author: Marc Haber <mh+debian-bugs@zugschlus.de>
Author: Colin Watson <cjwatson@debian.org>
Origin: other, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=542165#25
Bug-Debian: http://bugs.debian.org/542165
Forwarded: no
Last-Update: 2010-06-05

Index: grub-pc_2.00-SprezzOS2/util/grub-mkconfig_lib.in
===================================================================
--- grub-pc_2.00-SprezzOS2.orig/util/grub-mkconfig_lib.in	2012-06-27 11:27:53.000000000 -0400
+++ grub-pc_2.00-SprezzOS2/util/grub-mkconfig_lib.in	2012-09-30 11:10:26.890933369 -0400
@@ -129,6 +129,15 @@
     esac
   done
 
+  if dmsetup status $device 2>/dev/null | grep -q 'crypt[[:space:]]$'; then
+    grub_warn \
+      "$device is a crypto device, which GRUB cannot read directly.  Some" \
+      "necessary modules may be missing from /boot/grub/grub.cfg.  You may" \
+      "need to list them in GRUB_PRELOAD_MODULES in /etc/default/grub.  See" \
+      "http://bugs.debian.org/542165 for details."
+    return 0
+  fi
+
   # Abstraction modules aren't auto-loaded.
   abstraction="`"${grub_probe}" --device "${device}" --target=abstraction`"
   for module in ${abstraction} ; do
